docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,98 @@
+---
+title: add-discovery-kernel-arguments
+authors:
+- "@oamizur"
+  creation-date: 2022-09-20
+  last-updated: 2022-10-23
+---
+
+# Add discovery kernel arguments
+
+## Summary
+
+Some environments require additional kernel arguments to boot host machines.  
+We would like to support additional kernel arguments while booting the discovery image.
+
+## Motivation
+
+Previously, adding kernel arguments was required by several customers in order to adapt to various environments.
+For example **rd.net.timeout.carrier** may be required in order for either static networking to be applied or 
+to successfully get a dhcp address before pulling the rootfs.
+### Goals
+Add capability to set additional kernel arguments to discovery image.
+
+- Image types: Full ISO, minimal ISO, iPXE
+- Deployment types: Cloud, operator
+
+### Non-Goals
+Manipulation of existing kernel parameters (update/delete).  Update and delete of exiting
+kernel arguments in ISO file is supported by Coreos Installer, but this implementation will not support it. 
+
+## Proposal
+
+The discovery kernel arguments will be stored in the infra-env database record in a designated JSON encoded field.
```

Comment:
So if I need different args for different hosts, I'll need to create two infra envs?
That's a limitation. reasonable one I guess, but still a limitation.

Reply:
Worth stating explicitly.

Reply:
Done